qualtran/bloqs/mcmt/bloq_with_specialized_single_qubit_control.py

```diff
@@ -0,0 +1,102 @@
+#  Copyright 2024 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from typing import cast, Iterable, Optional, Protocol, runtime_checkable, Sequence, TYPE_CHECKING
+
+import numpy as np
+
+if TYPE_CHECKING:
+    from qualtran import AddControlledT, Bloq, BloqBuilder, CtrlSpec, SoquetT
+
+
+@runtime_checkable
+class BloqWithSpecializedControl(Protocol):
+    """mixin for a bloq that has a specialized single qubit controlled version."""
+
+    @property
+    def cv(self) -> Optional[int]: ...
+
+    def with_cv(self, *, cv: Optional[int]) -> Optional['Bloq']: ...
+
+    @property
+    def ctrl_reg_name(self) -> str: ...
+
+
+def get_ctrl_system_for_bloq_with_specialized_single_qubit_control(
+    bloq: 'BloqWithSpecializedControl', ctrl_spec: 'CtrlSpec'
+) -> tuple['Bloq', 'AddControlledT']:
+    from qualtran import Bloq, CtrlSpec, Soquet
+    from qualtran.bloqs.mcmt import ControlledViaAnd
+
+    def _get_default_fallback():
+        assert isinstance(bloq, Bloq)
+        return ControlledViaAnd.make_ctrl_system(bloq=bloq, ctrl_spec=ctrl_spec)
+
+    if ctrl_spec.num_qubits != 1:
+        return _get_default_fallback()
+
+    assert isinstance(
+        bloq, BloqWithSpecializedControl
+    ), f"{bloq} must implement protocol {BloqWithSpecializedControl}"
+
+    # extract the single bit control value
+    (cv,) = ctrl_spec._cvs_tuple
```

Comment:
I'm always confused by getting a particular control value from a CtrlSpec. The type annotation is really confusing because of the variety of valid `cv` values. Is it worth adding a property to CtrlSpec that returns a single bit control value (or errors if there are more than one)

Reply:
Yes this makes sense. Perhaps `CtrlSpec.get_single_control_bit`?

Reply:
added a method to extract the bit correctly